burnman/composite.py

```diff
@@ -33,63 +37,144 @@ class Composite(Material):
 
     This class is available as ``burnman.Composite``.
     """
-    def __init__(self, fractions, phases=None):
+    def __init__(self, phases, fractions=None, fraction_type='molar'):
         """
         Create a composite using a list of phases and their fractions (adding to 1.0).
 
         Parameters
         ----------
-        fractions: list of floats
-            molar fraction for each phase.
         phases: list of :class:`burnman.Material`
             list of phases.
+        fractions: list of floats
+            molar fraction for each phase.
         """
 
-        if phases is None:
-            # compatibility hack:
-            tmp = fractions
-            fractions = [pt[1] for pt in tmp]
-            phases = [pt[0] for pt in tmp]
-
-        assert(len(phases)==len(fractions))
+        
         assert(len(phases)>0)
+        self.phases = phases
+        
+        self.phase_names = []
+        for i in range(len(self.phases)):
+            try:
+                self.phase_names.append(self.phases[i].name)
```

Comment:
I don't like how names are handled with a try/except. We used to have to_string() which always works.